src/otx/algo/instance_segmentation/maskdino.py

```diff
@@ -0,0 +1,517 @@
+    mean = (123.675, 116.28, 103.53)
+    std = (58.395, 57.12, 57.375)
+
+    def _build_model(self, num_classes: int) -> nn.Module:
+        """Builds the model."""
+        return _MaskDINO.from_config(num_classes)
+
+    @property
+    def _exporter(self) -> OTXModelExporter:
+        """Creates OTXModelExporter object that can export the model."""
+        if self.input_size is None:
+            msg = f"Input size attribute is not set for {self.__class__}"
+            raise ValueError(msg)
+
+        return OTXNativeModelExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=(1, 3, *self.input_size),
+            mean=self.mean,
+            std=self.std,
+            resize_mode="standard",
+            swap_rgb=False,
+            via_onnx=True,
+            onnx_export_configuration={
+                "input_names": ["image"],
+                "dynamic_axes": {
+                    "image": {0: "batch", 2: "height", 3: "width"},
+                    "boxes": {0: "batch", 1: "num_dets"},
+                    "labels": {0: "batch", 1: "num_dets"},
+                    "masks": {0: "batch", 1: "num_dets", 2: "height", 3: "width"},
+                },
+                "opset_version": 16,
+                "autograd_inlining": False,
+            },
+            output_names=["bboxes", "labels", "masks"],
+        )
+
+    def configure_optimizers(self) -> tuple[list[torch.optim.Optimizer], list[dict[str, Any]]]:
+        """Configure an optimizer and learning-rate schedulers."""
+        param_groups = self._get_optim_params(self.model)
+        optimizer = self.optimizer_callable(param_groups)
+        optimizer_with_grad_clip = MaskDINOR50._add_grad_clipping(optimizer, clip_gradient_value=0.01)(
+            param_groups,
+            optimizer.defaults["lr"],
+        )
+
+        schedulers = self.scheduler_callable(optimizer_with_grad_clip)
+
+        def ensure_list(item: Any) -> list:  # noqa: ANN401
+            return item if isinstance(item, list) else [item]
+
+        lr_scheduler_configs = []
+        for scheduler in ensure_list(schedulers):
+            lr_scheduler_config = {"scheduler": scheduler}
+            if hasattr(scheduler, "interval"):
+                lr_scheduler_config["interval"] = scheduler.interval
+            if hasattr(scheduler, "monitor"):
+                lr_scheduler_config["monitor"] = scheduler.monitor
+            lr_scheduler_configs.append(lr_scheduler_config)
+
+        return [optimizer_with_grad_clip], lr_scheduler_configs
+
+    def _get_optim_params(self, model: nn.Module) -> list[dict[str, Any]]:
+        """Get optimizer parameters."""
+        _optimizer = self.optimizer_callable(self.parameters())
+
+        # Configurable from MaskDINO recipe
+        base_lr = _optimizer.defaults["lr"]
```

Comment:
User may use other configurations such as `eps` or etc.. Could you consider it?

Reply:
I agree that basic optimizer parameters (e.g., optimizer, learning rate, weight decay) should be configurable for now. However, I don't think it's necessary to expose advanced optimizer settings in this PR, as I'm not yet familiar with their impact on MaskDINO. Given the above reason, I suggest merging this PR as is and addressing optimization and fine-tuning in a later phase.